### ISSUE TITLE
in gatsby   window is not defined issue!

### ISSUE THREAD

**Issue by paolodedomini** [Grsmto/simplebar#428]
it seems an old issue but in the gatsby branch seems to remains 
error log:
`error Building static HTML failed

See our docs page on debugging HTML builds for help https://gatsby.dev/debug-html

  22 | var cachedScrollbarWidth = null;
  23 | var cachedDevicePixelRatio = null;
> 24 | window.addEventListener('resize', function () {
     | ^
  25 |   if (cachedDevicePixelRatio !== window.devicePixelRatio) {
  26 |     cachedDevicePixelRatio = window.devicePixelRatio;
  27 |     cachedScrollbarWidth = null;

  WebpackError: ReferenceError: window is not defined`

**Comment by paolodedomini** [comment]
this error comes out just at build time.
In develop mode there's no problems at all
thanx

**Comment by shavidzet** [comment]
So the problem is, that SimpleBar module tries to access `window` object, which is part of DOM and it's only available in browser, not in node env and while gatsby tries to export template on SSR it shows error.

My custom/temporary solution is to create wrapper component for SimpleBar to import dependency client-side only:

SimpleBar.js
```js
import React from 'react'
import 'simplebar/dist/simplebar.min.css'

const Component = (props) => {
  if (typeof window === 'undefined') {
    return false
  }

  const SimpleBar = require('simplebar-react').default
  return (
    <SimpleBar {...props} />
  )
}

export default Component
```

You can use this component instead of directly importing `simplebar` package...

**Comment by Grsmto** [comment]
Are you sure you're using the latest version? 
Every calls to `window` are protected for SSR so you should not face this issue. Actually the site of this plugin itself is using Gatsby.

**Comment by diegopego86** [comment]
Hi @shavidzet ,I was having the same issue and after reading other git issues and testing different versions I found the one that works for me:

Initially I tried the one suggested here: https://github.com/Grsmto/simplebar/issues/396 but didn't work for me then after seeing this post, I looked up for the latest release and I found this one.

simplebar-react@2.0.6-corejs2

I was able to run gatsby build with no issues.

Thanks @Grsmto for all  the hardwork!
